scripts/screens/ChangeGenderScreen.py

```diff
@@ -0,0 +1,485 @@
+                        ujson.dump(pronouns_dict, file, indent=2)
+                    self.update_custom_preset_removal()
+                    #print("After update:", pronouns_dict["custom_presets"])
+
+            else:
+                if len(self.removalbuttons) == len(self.the_cat.pronouns):
+                    for pronounset in self.the_cat.pronouns:
+                        remove_button_id = f"remove_button_{pronounset['name']}"
+                        if event.ui_element == self.removalbuttons[remove_button_id]:
+                            #print(f"Remove button clicked for current pronoun: {pronounset['name']}")
+                            self.remove_pronounset(pronounset['name'])
+                
+            with open('resources/dicts/pronouns.json', 'r', encoding='utf-8') as file:
+                    pronouns_data = ujson.load(file)
+
+            for pronounset in pronouns_data["custom_presets"]:
+                remove_button_id = f"remove_button_{pronounset['name']}2"
+                if event.ui_element == self.buttons.get(remove_button_id):
+                    #print(f"Remove button clicked for pronoun preset: {pronounset['name']}")
+                    pronouns_data["custom_presets"].remove(pronounset)
+                    with open('resources/dicts/pronouns.json', 'w', encoding='utf-8') as file:
+                        ujson.dump(pronouns_data, file, ensure_ascii=False, indent=4)
+                    self.remove_custom_preset(pronounset['name'])
+
+                
+        elif event.type == pygame.KEYDOWN and game.settings['keybinds']:
+            if event.key == pygame.K_ESCAPE:
+                self.change_screen("profile screen")
+            elif event.key == pygame.K_RIGHT:
+                game.switches["cat"] = self.next_cat
+                self.update_selected_cat()
+            elif event.key == pygame.K_LEFT:
+                game.switches["cat"] = self.previous_cat
+                self.update_selected_cat()
+
+    def screen_switches(self):
+
+        self.next_cat_button = UIImageButton(scale(pygame.Rect((1244, 50), (306, 60))), "", object_id="#next_cat_button"
+                                             , manager=MANAGER)
+        self.previous_cat_button = UIImageButton(scale(pygame.Rect((50, 50), (306, 60))), "",
+                                                 object_id="#previous_cat_button"
+                                                 , manager=MANAGER)
+        self.back_button = UIImageButton(scale(pygame.Rect((50, 120), (210, 60))), "", object_id="#back_button"
+                                         , manager=MANAGER)
+        self.update_selected_cat()
+    
+    '''def on_remove_button_click(self, pronoun_name):
+        print(f"Remove button clicked for pronoun: {pronoun_name}")'''
+        
+    def get_sample_text(self, pronouns):
+        text = ""
+        text += f"Demo: {pronouns['name']} <br>"
```

Comment:
Game in currently crashing on this line with a throw `Keyerror` when this page attempts to open. This only seems to occur on load-in clans.